### PR TITLE
batch update of typings to use `HTMLPropsFor`

### PR DESCRIPTION
**Pull request by danethurber**
TODO:
- [x] appframe
- [x] avatar
- [x] badge
- [x] banner
- [x] breadcrumb 
- [x] button
- [x] card
- [x] carousel
- [x] checkbox
- [x] circularprogress
- [x] core
- [x] datawell
- [x] datepicker
- [x] dialog
- [x] drawer
- [x] dropdown

### CODE REVIEW

File: packages/avatar/package.json [hunk 1]
```diff
@@ -15,13 +15,14 @@
...
     "storybook": "start-storybook -p 6006",
-    "test": "jest",
+    "test": "tsc --noEmit --project ./tsconfig.json && jest",
     "test:watch": "yarn test -- --watchAll"
   },
   "style": "dist/index.css",
   "types": "dist/esm/index.d.ts",
   "dependencies": {
     "@pluralsight/ps-design-system-core": "^6.5.1",
+    "@pluralsight/ps-design-system-util": "*",
```
Comment (jaketrent) [inline packages/avatar/package.json:25]: I still feel wierd marking * in the dependencies.  Don't we want some version ranging?

File: packages/avatar/tsconfig.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
   "compilerOptions": {
     "allowJs": true,
     "rootDir": null,
-    "jsx": "react"
+    "jsx": "react",
+    "strict": true
```
Comment (jaketrent) [inline packages/avatar/tsconfig.json:7]: Man, you got ones that needed a fair bit more tsc love.  Sorry! :/

File: packages/banner/src/react/index.tsx [hunk 3]
```diff
@@ -64,19 +65,20 @@ const Banner = React.forwardRef((props, ref) => {
...
-  > {}
-
-const Button = React.forwardRef((props, forwardRef) => {
+interface AnchorProps extends HTMLPropsFor<'a'> {
+  href: string
+}
+interface ButtonProps extends HTMLPropsFor<'button'> {
+  href?: undefined
+}
+type ButtonComponent = React.ForwardRefExoticComponent<unknown> & {
```
Comment (jaketrent) [inline packages/banner/src/react/index.tsx:74]: Are there other components outside of this PR that need their overloads updated to be like this?

File: packages/button/src/react/index.tsx [hunk 4]
```diff
@@ -136,109 +132,111 @@ interface BaseButtonProps {
...
+          style={style}
+        >
+          {iconEl}
+          {labelEl}
+        </button>
+      )
+    }
   }
-}) as ButtonComponent
+) as ButtonComponent & ButtonStatics
```
Comment (jaketrent) [inline packages/button/src/react/index.tsx:239]: I also like that this approach uses the React typings directly w/o needing our wrapper.

**Review by jaketrent — APPROVED** [review]
Typings look good to me.  

Local build failing on what seems like Banner.  Might be an adjustment needed there.

```
lerna ERR! yarn run build exited 1 in '@pluralsight/ps-design-system-banner'
lerna ERR! yarn run build stdout:
$ run-s build:cjs build:esm build:css
$ tsc --project tsconfig.build.json --module commonjs --target es5 --outDir dist/cjs
src/react/index.tsx(85,5): error TS2769: No overload matches this call.
  Overload 1 of 3, '(props: unknown): ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<any, any, any>)> | null', gave the following error.
    Type '{ ref: any; appearance: string; size: string; href: string; key?: string | number | null | undefined; download?: any; hrefLang?: string | undefined; media?: string | undefined; ping?: string | undefined; ... 257 more ...; onTransitionEndCapture?: ((event: TransitionEvent<...>) => void) | undefined; } | { ...; }' is not assignable to type 'IntrinsicAttributes'.
      Type '{ ref: any; appearance: string; size: string; href: string; key?: string | number | null | undefined; download?: any; hrefLang?: string | undefined; media?: string | undefined; ping?: string | undefined; ... 257 more ...; onTransitionEndCapture?: ((event: TransitionEvent<...>) => void) | undefined; }' is not assignable to type 'IntrinsicAttributes'.
        Property 'ref' does not exist on type 'IntrinsicAttributes'.
  Overload 2 of 3, '(props: ButtonAnchorProps, ref?: RefFor<"a"> | undefined): Element', gave the following error.
    Type '{ ref: any; appearance: string; size: string; href: string; key?: string | number | null | undefined; download?: any; hrefLang?: string | undefined; media?: string | undefined; ping?: string | undefined; ... 257 more ...; onTransitionEndCapture?: ((event: TransitionEvent<...>) => void) | undefined; } | { ...; }' is not assignable to type 'IntrinsicAttributes & ButtonAnchorProps'.
      Type '{ ref: any; appearance: string; size: string; href?: undefined; key?: string | number | null | undefined; autoFocus?: boolean | undefined; disabled?: boolean | undefined; form?: string | undefined; ... 261 more ...; onTransitionEndCapture?: ((event: TransitionEvent<...>) => void) | undefined; }' is not assignable to type 'ButtonAnchorProps'.
        Types of property 'href' are incompatible.
          Type 'undefined' is not assignable to type 'string'.
  Overload 3 of 3, '(props: ButtonButtonProps, ref?: RefFor<"button"> | undefined): Element', gave the following error.
    Type '{ ref: any; appearance: string; size: string; href: string; key?: string | number | null | undefined; download?: any; hrefLang?: string | undefined; media?: string | undefined; ping?: string | undefined; ... 257 more ...; onTransitionEndCapture?: ((event: TransitionEvent<...>) => void) | undefined; } | { ...; }' is not assignable to type 'IntrinsicAttributes & ButtonButtonProps'.
      Type '{ ref: any; appearance: string; size: string; href: string; key?: string | number | null | undefined; download?: any; hrefLang?: string | undefined; media?: string | undefined; ping?: string | undefined; ... 257 more ...; onTransitionEndCapture?: ((event: TransitionEvent<...>) => void) | undefined; }' is not assignable to type 'ButtonButtonProps'.
        Types of property 'href' are incompatible.
          Type 'string' is not assignable to type 'undefined'.
```

File: packages/avatar/package.json (hunk 1 above)
Comment (danethurber) [reply packages/avatar/package.json:25]: when we publish it will add the version range :)

File: packages/banner/src/react/index.tsx (hunk 3 above)
Comment (danethurber) [reply packages/banner/src/react/index.tsx:74]: i think NavUser needs the `unknown`...but that's about it

**Pull request merged**